lms/djangoapps/ccx/api/v0/views.py

```diff
@@ -0,0 +1,651 @@
+    serializer_class = CCXCourseSerializer
+    pagination_class = CCXAPIPagination
+
+    def get(self, request):
+        """
+        Gets a list of CCX Courses for a given Master Course.
+
+        Additional parameters are allowed for pagination purposes.
+
+        Args:
+            request (Request): Django request object.
+
+        Return:
+            A JSON serialized representation of a list of CCX courses.
+        """
+        master_course_id = request.GET.get('master_course_id')
+        master_course_object, master_course_key, error_code, http_status = get_valid_course(master_course_id)
+        if master_course_object is None:
+            return Response(
+                status=http_status,
+                data={
+                    'error_code': error_code
+                }
+            )
+
+        queryset = CustomCourseForEdX.objects.filter(course_id=master_course_key)
+        order_by_input = request.query_params.get('order_by')
+        sort_order_input = request.query_params.get('sort_order')
+        if order_by_input in ('id', 'display_name'):
+            sort_direction = ''
+            if sort_order_input == 'desc':
+                sort_direction = '-'
+            queryset = queryset.order_by('{0}{1}'.format(sort_direction, order_by_input))
+        page = self.paginate_queryset(queryset)
+        serializer = self.get_serializer(page, many=True)
+        response = self.get_paginated_response(serializer.data)
+        return response
+
+    def post(self, request):
+        """
+        Creates a new CCX course for a given Master Course.
+
+        Args:
+            request (Request): Django request object.
+
+        Return:
+            A JSON serialized representation a newly created CCX course.
+        """
+        master_course_id = request.data.get('master_course_id')
+        master_course_object, master_course_key, error_code, http_status = get_valid_course(
+            master_course_id,
+            advanced_course_check=True
+        )
+        if master_course_object is None:
+            return Response(
+                status=http_status,
+                data={
+                    'error_code': error_code
+                }
+            )
+
+        # validating the rest of the input
+        valid_input, field_errors = get_valid_input(request.data)
+        if field_errors:
+            return Response(
+                status=status.HTTP_400_BAD_REQUEST,
+                data={
+                    'field_errors': field_errors
+                }
+            )
+
+        try:
+            coach = User.objects.get(email=valid_input['coach_email'])
+        except User.DoesNotExist:
+            return Response(
+                status=status.HTTP_404_NOT_FOUND,
+                data={
+                    'error_code': 'coach_user_does_not_exist'
+                }
+            )
+
+        # TODO: verify that the transaction.atomic is enough for EDX
```

Comment:
Did you verify? :)

So - either `ATOMIC_REQUESTS` will be True or False.
* If True, then this view will be wrapped in an atomic transaction automatically and this context manager will start a checkpoint. It appears all the write operations are contained within this block, so they'll all either be added to the checkpoint and transaction if the block completes with no errors *or* they'll be rolled back from the checkpoint together.
* If False, then this context manager will start a transaction - and all the write operations will be committed or rolled back at the end of the block.

Either way, seems like the desired result is accomplished. So the comment can be removed - unless there are other considerations?